src/main/java/br/ufpe/cin/mergers/handlers/MethodAndConstructorRenamingAndDeletionHandler.java

```diff
@@ -133,18 +135,25 @@ private FSTNode getCorrespondentNode(FSTNode node, FSTNode tree) {
     }
 
     private FSTNode getMostAccurateMatch(FSTNode node, FSTNode contributionTree) {
-        for(FSTNode contributionNode : Traverser.collectTerminals(contributionTree)) {
-            if(RenamingUtils.isMethodOrConstructorNode(contributionNode) && areVerySimilarNodes(node, contributionNode))
-                return contributionNode;
+        List<FSTTerminal> terminals = Traverser.collectTerminals(contributionTree);
```

Comment:
@jvcoutinho no exemplo, houve o matching de  add(int a, int b) não modificado por left com o addIntegers de right, certo? Isso teria sido evitado se essa lista de terminais contivesse apenas os nós terminais editados, em vez de todos os terminais. A classe MergeContext já não tinha algum atributo que serveria a esse propósito?

Reply:
Mas como uma das condições é se a assinatura for a mesma, acaba não tendo necessidade. Além disso, só comparamos nós da base que foram renomeados ou deletados por alguma das contribuições.

O que aconteceu foi um pouco mais complexo. Todos os métodos (`add`, `sub`, `mul` e `div`) da base foram pareados com `add` de left e `addIntegers` de right pela 3ª condição (corpos similares e assinaturas iguais exceto pelo nome). Desse modo, o handler interpretou o match de `div` (o último match) com `add` e `div` com `addIntegers` como um mutual renaming para assinaturas diferentes -> conflito.

Pelo novo sistema, todos eles tiveram match correto graças à 2ª condição (corpos iguais).